oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,140 @@
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.jose.jws.JwsAlgorithms;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtVerifier;
+
+import javax.crypto.SecretKey;
+import java.security.PublicKey;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+public class OAuth2ResourceServerConfigurer<B extends HttpSecurityBuilder<B>> extends
+		AbstractHttpConfigurer<JwtVerifierConfigurer<B>, B> {
+
+	public BearerConfigurer bearer() {}
+
+	public class BearerConfigurer {
+		public BearerConfigurer inBody() {}
+
+		public BearerConfigurer inHeader() {}
+
+		public BearerConfigurer inQuery() {}
+
+		public OAuth2ResourceServerConfigurer<B> and() {}
+	}
+
+
+
+	public JwtConfigurer jwt() {}
+
+	public class JwtConfigurer {
+		public JwtDecoderConfigurer decoder() {}
+
+		public JwtConfigurer decoder(JwtDecoder jwtDecoder) {}
+
+		public JwtVerifierConfigurer verifier() {}
+
+		public JwtVerifier verifier(JwtVerifier jwtVerifier) {}
+
+		public OAuth2ResourceServerConfigurer<B> and() {}
+	}
+
+	public class JwtDecoderConfigurer {
+		public JwtDecoderConfigurer rs256() {}
+
+		public JwtDecoderConfigurer algorithm(JwsAlgorithms algorithm) {}
+
+		public JwkSetConfigurer jwkSet() {}
+
+		public JwtConfigurer and() {}
+	}
+
+	public class JwtVerifierConfigurer {
```

Comment:
Providing specific configuration points for claims can get fairly large given that there are quite a few claims defined in JWT, OIDC, etc... I believe we can provide something like this to simplify:

`http.oauth2().resourceServer().accessTokenVerifiers(OAuth2TokenVerifier... verifiers)`

This configuration point can also be reused for other Access Token formats, for example, Opaque (Introspection Endpoint).

Reply:
Sounds good, I'll just refactor the `verifier` method and push it up a level. Makes sense.